### PR TITLE
fix(SCT-1082): large response size getting submissions

### PR DESCRIPTION
**Pull request by JohnFarrellDev**
## Link to JIRA ticket

[Ticket](https://hackney.atlassian.net/browse/SCT-1082)

### *What is the problem we're trying to solve*

Response size from GET /api/v1/submissions is way too large.

### *What changes have we introduced*

By default do not include EditHistory or FormAnswers when getting a list of case submissions to reduce JSON response size.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [x] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### *Follow up actions after merging PR*

Associated frontend changes

### CODE REVIEW

File: SocialCareCaseViewerApi/V1/Factories/EntityFactory.cs [hunk 1]
```diff
@@ -156,14 +156,16 @@ public static Domain.CaseSubmission ToDomain(this CaseSubmission caseSubmission)
...
                     EditTime = e.EditTime,
                     Worker = e.Worker.ToDomain(false)
-                }).ToList(),
+                }).ToList() : null,
                 SubmissionState = mapSubmissionStateToString[caseSubmission.SubmissionState],
-                FormAnswers = caseSubmission.FormAnswers,
-                Title = caseSubmission.Title
+                FormAnswers = includeFormAnswers ? caseSubmission.FormAnswers : null,
+                Title = caseSubmission.Title,
+                LastEdited = caseSubmission.EditHistory.Last().EditTime,
```
Comment (LBHTKarki) [inline SocialCareCaseViewerApi/V1/Factories/EntityFactory.cs:167]: Will initial submission have LastEdited value? So the .Last() will always have a value

Comment (JohnFarrellDev) [reply SocialCareCaseViewerApi/V1/Factories/EntityFactory.cs:167]: I'm tempted to make it nullable but based on FormUseCase#ExecutePost the initially created CaseSubmission does start with a single EditHistory item. .Last() should always return an item.

**Comment by LBHTKarki** [comment]
Would be interesting to investigate doing the field filtering in the db query. That would speed up the data load quite a bit I imagine

**Comment by JohnFarrellDev** [comment]
> Would be interesting to investigate doing the field filtering in the db query. That would speed up the data load quite a bit I imagine

It's a good point, I'll look into if there is a way to make the request to not even ask for the properties not required

**Comment by martapederiva** [comment]
> > Would be interesting to investigate doing the field filtering in the db query. That would speed up the data load quite a bit I imagine
> 
> It's a good point, I'll look into if there is a way to make the request to not even ask for the properties not required

Would love to help on this (DBs ❤️ )

**Review by LBHTKarki — APPROVED** [review]

**Comment by JohnFarrellDev** [comment]
> Would be interesting to investigate doing the field filtering in the db query. That would speed up the data load quite a bit I imagine

For now we're not going to do this. Reason is that to get LastEdited property on our domain object we require the EditHistory, to get CompletedSteps on our domain object we require FormAnswers.

It may be possible in the future to make this improvement but for now we don't consider it worth the time required.

**Pull request merged**